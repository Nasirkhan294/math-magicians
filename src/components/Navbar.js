import { Link } from 'react-router-dom';

const Navbar = () => (
  <div className="navbar">
    <h1>~ Math Magician ~</h1>
    <ul>
      <li>
        <Link to="/" className="nav-link">Home</Link>
      </li>
      <li>
        <Link to="/Calculator" className="nav-link">Calculator</Link>
      </li>
      <li>
        <Link to="/Quote" className="nav-link">Quote</Link>
      </li>
    </ul>
  </div>
);

export default Navbar;
