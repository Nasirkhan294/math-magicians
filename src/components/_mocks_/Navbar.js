import React from 'react';
import { BrowserRouter, NavLink } from 'react-router-dom';

const Navbar = [
  { to: '/', text: 'Home' },
  { to: '/calculator', text: 'Calculator' },
  { to: '/quote', text: 'Quote' },
];

const Nav = () => (
  <BrowserRouter>
    <nav className="navbar">
      <h1>~ Math Magician ~</h1>
      <ul>
        {Navbar.map((link) => (
          <li key={link.to}>
            <NavLink to={link.to} className="nav-link">
              {link.text}
            </NavLink>
          </li>
        ))}
      </ul>
    </nav>
  </BrowserRouter>
);

export default Nav;
