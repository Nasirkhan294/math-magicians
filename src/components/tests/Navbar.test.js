import React from 'react';
import { render } from '@testing-library/react';
import Navbar from '../_mocks_/Navbar';

test('Navbar matches snapshot', () => {
  const { asFragment } = render(<Navbar />);
  expect(asFragment()).toMatchSnapshot();
});
