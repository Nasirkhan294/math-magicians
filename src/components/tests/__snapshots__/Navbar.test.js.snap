// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="navbar"
  >
    <h1>
      ~ Math Magician ~
    </h1>
    <ul>
      <li>
        <a
          aria-current="page"
          class="nav-link active"
          href="/"
        >
          Home
        </a>
      </li>
      <li>
        <a
          class="nav-link"
          href="/calculator"
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          class="nav-link"
          href="/quote"
        >
          Quote
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
